File: api/src/Domain/Settings/Entities/Business.cs
using System;

namespace DetailingArsenal.Domain.Settings {
    public class Business : Aggregate<Business>, IUserEntity {
        public const int NameMaxLength = 64;
        public const int AddressMaxLength = 128;
        public const int PhoneMaxLength = 32;

        public Guid UserId { get; set; } = Guid.Empty;

        /// <summary>
        /// Actual business name.
        /// </summary>
        public string? Name {
            get => name;
            set {
                if (value?.Length > NameMaxLength) {
                    throw new ArgumentOutOfRangeException();
                }

                name = value;
            }
        }

        /// <summary>
        /// Physical address of the business.
        /// </summary>
        public string? Address {
            get => address;
            set {
                if (value?.Length > AddressMaxLength) {
                    throw new ArgumentOutOfRangeException();
                }

                address = value;
            }
        }

        /// <summary>
        /// Contact phone number of the business.
        /// </summary>
        public string? Phone {
            get => phone;
            set {
                if (value?.Length > PhoneMaxLength) {
                    throw new ArgumentOutOfRangeException();
                }

                phone = value;
            }
        }

        private string? name = null!;
        private string? address = null;
        private string? phone = null;

        public static Business Create(Guid userId) {
            return new Business() {
                Id = Guid.NewGuid(),
                UserId = userId
            };
        }
    }
}

File: app/src/api/scheduling/settings/data-transfer-objects/business.ts
/**
 * Business information such as the name, address, and phone number.
 */
export class Business {
    public static NAME_MAX_LENGTH = 64;
    public static ADDRESS_MAX_LENGTH = 128;
    public static PHONE_MAX_LENGTH = 32;

    /**
     * Create a new business entity.
     * @param name The official name of the business.
     * @param address The street address.
     * @param phone The contact phone number.
     */
    constructor(public id: string, public name?: string, public address?: string, public phone?: string) {
        if (name != null && name.length > Business.NAME_MAX_LENGTH) {
            throw new RangeError('name');
        }

        if (address != null && address.length > Business.ADDRESS_MAX_LENGTH) {
            throw new RangeError('address');
        }

        if (phone != null && phone.length > Business.PHONE_MAX_LENGTH) {
            throw new RangeError('phone');
        }
    }
}
